src/groupeddataframe/groupeddataframe.jl

```diff
@@ -387,6 +391,29 @@ function Base.to_index(gd::GroupedDataFrame, key::NamedTuple{N}) where {N}
     return Base.to_index(gd, Tuple(key))
 end
 
+function _dict_to_tuple(key::AbstractDict{<:AbstractString} , gd)
+    T = keytype(key)
+    return ntuple(length(gd.cols)) do i 
+        # Most `<: AbstractString` types do not support constructors from symbols
+        key[T(string(gd.cols[i]))]
+    end
+end
+
+function _dict_to_tuple(key::AbstractDict{Symbol}, gd)
+    T = keytype(key)
+    return ntuple(length(gd.cols)) do i 
+        key[T(gd.cols[i])]
+    end
+end
+
+function Base.to_index(gd::GroupedDataFrame, key::Union{AbstractDict{Symbol},AbstractDict{<:AbstractString}})
+    if length(key) != length(gd.cols)
+        throw(KeyError(key))
```

Comment:
if you drop this check here (in favor of `_dict_to_tuple` - this will be a simpler one liner)